erpnext/hr/doctype/leave_ledger_entry/leave_ledger_entry.py

```diff
@@ -183,4 +183,64 @@ def expire_carried_forward_allocation(allocation):
 			from_date=allocation.to_date,
 			to_date=allocation.to_date
 		)
-		create_leave_ledger_entry(allocation, args)
\ No newline at end of file
+		create_leave_ledger_entry(allocation, args)
+
+
+def get_leave_allocation(employee, leave_type, date, fields=None):
+	allocation = frappe.db.get_value('Leave Allocation',
+		filters={
+			'employee': employee,
+			'leave_type': leave_type,
+			'from_date': ['<=', date],
+			'to_date': ['>=', date],
+			'docstatus': 1
+		},
+		fieldname=fields or "name",
+		as_dict=True if isinstance(fields, list) else False
+	)
+	if not allocation:
+		frappe.throw(_("Leave Allocation not found."))
+
+	return allocation
+
+
+def delete_expired_leave_ledger_entry(leave_allocation):
+	if not leave_allocation:
+		return
+
+	to_date = frappe.db.get_value("Leave Allocation", leave_allocation, "to_date")
+	if not to_date or getdate() <= to_date:
+		return
+
+	name = frappe.get_value(
+		'Leave Ledger Entry',
+		filters={
+			"transaction_type": "Leave Allocation",
+			"transaction_name": leave_allocation,
+			"is_expired": 1
+		}
+	)
+	if name:
+		frappe.db.sql("""DELETE FROM `tabLeave Ledger Entry` WHERE `name`=%s""", (name))
+
+
+def get_allocation_leave_balance_summary(allocation_name):
+	allocation = frappe.get_doc("Leave Allocation", allocation_name)
+
+	leaves_summary = frappe.db.sql("""
+		SELECT
+			SUM(leaves) as used_leaves
+		FROM `tabLeave Ledger Entry`
+		WHERE
+			employee = %(employee)s
+			AND leave_type = %(leave_type)s
+			AND docstatus = 1
+			AND leaves < 0
+			AND from_date between %(from_date)s AND %(to_date)s
```

Comment:
Fix the date condition. Use the date overlap condition instead of this

Reply:
using get_leaves_for_period from leave_application